blocks/rich-text/index.js

```diff
@@ -638,8 +639,13 @@ export class RichText extends Component {
 
 			const beforeElement = nodeListToReact( beforeFragment.childNodes, createTinyMCEElement );
 			const afterElement = nodeListToReact( filterEmptyNodes( afterFragment.childNodes ), createTinyMCEElement );
+
+			// Since we need to abort the change done by TinyMCE
+			// we need to call updateContent to restore the initial content before calling onSplit.
+			this.updateContent();
```

Comment:
If it's the case that we must call `updateContent` before every `this.props.onSplit`, could be wise to create a separate class function for consistency/safety's sake.